programs/perpetuals/src/state/pool.rs

```diff
@@ -330,8 +330,8 @@ impl Pool {
     }
 
     pub fn get_interest_amount_usd(&self, position: &Position, custody: &Custody) -> Result<u64> {
-        let rate_diff = if custody.borrow_rate_sum > position.borrow_rate_sum {
-            math::checked_sub(custody.borrow_rate_sum, position.borrow_rate_sum)? as u128
+        let rate_diff = if custody.borrow_rate_state.rate_sum > position.borrow_rate_sum {
+            math::checked_sub(custody.borrow_rate_state.rate_sum, position.borrow_rate_sum)? as u128
```

Comment:
ahh so this is what the user is owning the pool?

Reply:
so a user needs to pay the `rate_diff` only when the position is `closed` or when there is a liquidation?

Reply:
so, we compute a running sum of the borrow rate, initial value of that sum is recorded upon position open. when position is about to close we take a difference between current sum and initial value and this is how much the user has to pay.

Reply:
In the case of dynamic borrowing rates the rate_sum should be updated everything the borrowing rate is updated.

Reply:
this has been addressed